caom2-tap/src/intTest/java/ca/nrc/cadc/caom2ops/CaomTapQueryTest.java

```diff
@@ -0,0 +1,295 @@
+{
+    private static final Logger log = Logger.getLogger(CaomTapQueryTest.class);
+    
+    private static final URI OBS_URI = URI.create("caom:IRIS/f212h000");
+    private static final URI PLANE_URI = URI.create("caom:IRIS/f212h000/IRAS-60um");
+    private static final URI PUB_ID = URI.create("ivo://cadc.nrc.ca/IRIS?f212h000/IRAS-60um");
+    
+    private static final String TAP_URI = "ivo://cadc.nrc.ca/tap";
+    private final DateFormat dateFormat = DateUtil.getDateFormat(DateUtil.IVOA_DATE_FORMAT, DateUtil.UTC);
+    
+    static
+    {
+        Log4jInit.setLevel("ca.nrc.cadc.caom2ops", Level.INFO);
+    }
+
+    //@Test
+    public void testTemplate()
+    {
+        try
+        {
+
+        }
+        catch(Exception unexpected)
+        {
+            log.error("unexpected exception", unexpected);
+            Assert.fail("unexpected exception: " + unexpected);
+        }
+    }
+
+    @Test
+    public void testConstructor()
+    {
+        try
+        {
+            String runID = "testJobID";
+	    URI tapURI = new URI(TAP_URI);
+            
+            // minimal args
+            CaomTapQuery query = new CaomTapQuery(tapURI, null);
+            
+            // all args
+	    query = new CaomTapQuery(tapURI, runID);
+
+            try
+            {
+                // missing required arg
+                query = new CaomTapQuery(null, "foo");
+                Assert.fail("expected IllegalArgumentException for tapURL=null");
+            }
+            catch(IllegalArgumentException expected) { log.debug("expected: " + expected); }
+        }
+        catch(Exception unexpected)
+        {
+            log.error("unexpected exception", unexpected);
+            Assert.fail("unexpected exception: " + unexpected);
+        }
+    }
+
+    @Test
+    public void testObservationQuery()
+    {
+        try
+        {
+            ObservationURI uri = new ObservationURI(OBS_URI);
+            CaomTapQuery query = new CaomTapQuery(URI.create(TAP_URI), "CaomTapQueryTest");
+            Observation o = query.performQuery(uri);
+            Assert.assertNotNull(o);
+            Assert.assertEquals(uri.getCollection(), o.getCollection());
+            Assert.assertEquals(uri.getObservationID(), o.getObservationID());
+            
+            ObservationWriter w = new ObservationWriter();
+            w.write(o, new FileWriter("CaomTapQueryTest.xml"));
+            
+            // this is needed to diagnose which plane has checksum mismatch
+            //for (Plane p : o.getPlanes())
+            //{
+            //    URI metaChecksum = p.computeMetaChecksum(MessageDigest.getInstance("MD5"));
+            //    Assert.assertEquals("Plane.metaChecksum", p.getMetaChecksum(), metaChecksum);
+            //    
+            //    URI accMetaChecksum = p.computeAccMetaChecksum(MessageDigest.getInstance("MD5"));
+            //    Assert.assertEquals("Plane.accMetaChecksum", p.getAccMetaChecksum(), accMetaChecksum);
+            //}
```

Comment:
Should be enabled?

Reply:
this goes along with the circles in position_bounds_points: by having tap output the position_bounds column instead of position_bounds_points, the trip through tap becomes lossy so I had to disable the the metaChecksum tests.

Reply:
Ah! Understood.